### PR TITLE
Add AKS config for review apps

### PR DESCRIPTION
**Pull request by peteryates**
Follows on from #3587

### CODE REVIEW

**Comment by peteryates** [comment]
https://trello.com/c/mmO6LoNM/313-ecf-allow-review-apps-to-be-deployed-to-aks

File: Makefile [hunk 1]
```diff
@@ -87,6 +95,34 @@ domains-apply: domains-init # make dev domains-apply
...
+		--parameters "resourceGroupName=${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-rg" 'tags=${RG_TAGS}' \
+			"tfStorageAccountName=${AZURE_RESOURCE_PREFIX}${SERVICE_SHORT}tfstate${CONFIG_SHORT}sa" "tfStorageContainerName=${SERVICE_SHORT}-tfstate" \
+			keyVaultNames='("${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-app-kv", "${AZURE_RESOURCE_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-inf-kv")' \
+			"enableKVPurgeProtection=${KEY_VAULT_PURGE_PROTECTION}" ${WHAT_IF}
+.PHONY: ci
+ci:	## Run in automation environment
+	$(eval SP_AUTH=true)
+	# $(eval DISABLE_PASSCODE=true)
+	$(eval AUTO_APPROVE=-auto-approve)
+	# $(eval CONFIRM_PRODUCTION=true)
```
Comment (peteryates) [inline Makefile:107]: Do we need to re-add these commented out lines? 🤔

File: terraform/aks/application.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  environment  = local.environment
+  service_name = local.service_name
+
+  cluster_configuration_map = module.cluster_data.configuration_map
+
+  kubernetes_config_map_name = module.application_configuration.kubernetes_config_map_name
+  kubernetes_secret_name     = module.application_configuration.kubernetes_secret_name
+
+  docker_image = var.docker_image
+}
```
Comment (saliceti) [inline terraform/aks/application.tf:53]: By default the module will use /healthcheck as probe. This should be the simplest one possible, which corresponds to your /check. So I suggest using `probe_path = "/check"` here.

File: config/environments/deployed_development.rb [hunk 3]
```diff
@@ -44,7 +44,8 @@
   config.active_storage.service = :amazon
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  config.force_ssl = true
+  config.ssl_options = { exclude: ->(request) { request.path.include?("healthcheck") } }
```
Comment (saliceti) [inline config/environments/deployed_development.rb:47]: As per https://github.com/DFE-Digital/early-careers-framework/pull/3588/files#r1262781810 change to "check" here

**Review by saliceti — APPROVED** [review]
LGTM!

**Pull request merged**